streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/FlinkClientHandler.scala

```diff
@@ -25,55 +25,55 @@ object FlinkClientHandler {
 
   def submit(submitInfo: SubmitRequest): SubmitResponse = {
     submitInfo.executionMode match {
-      case ExecutionMode.LOCAL => LocalSubmit.submit(submitInfo)
-      case ExecutionMode.REMOTE => RemoteSubmit.submit(submitInfo)
-      case ExecutionMode.YARN_APPLICATION => YarnApplicationSubmit.submit(submitInfo)
-      case ExecutionMode.YARN_SESSION => YarnSessionSubmit.submit(submitInfo)
-      case ExecutionMode.YARN_PER_JOB => YarnPerJobSubmit.submit(submitInfo)
-      case ExecutionMode.KUBERNETES_NATIVE_SESSION => KubernetesNativeSessionSubmit.submit(submitInfo)
-      case ExecutionMode.KUBERNETES_NATIVE_APPLICATION => KubernetesNativeApplicationSubmit.submit(submitInfo)
+      case ExecutionMode.LOCAL => LocalClient.submit(submitInfo)
+      case ExecutionMode.REMOTE => RemoteClient.submit(submitInfo)
+      case ExecutionMode.YARN_APPLICATION => YarnApplicationClient.submit(submitInfo)
+      case ExecutionMode.YARN_SESSION => YarnSessionClient.submit(submitInfo)
+      case ExecutionMode.YARN_PER_JOB => YarnPerJobClient.submit(submitInfo)
+      case ExecutionMode.KUBERNETES_NATIVE_SESSION => KubernetesNativeSessionClient.submit(submitInfo)
+      case ExecutionMode.KUBERNETES_NATIVE_APPLICATION => KubernetesNativeApplicationClient.submit(submitInfo)
       case _ => throw new UnsupportedOperationException(s"Unsupported ${submitInfo.executionMode} submit ")
```

Comment:
There's some steal condition in the client.
1. The ExecutionMode field is required for every existed XXXRequest and behavior like these lines.
2. every request need a method to drive the action bounded with `xxxClient`
3. special ExecutionMode with special xxxClient.

If we want to remove the case lines , which was mentioned in the `trigger savepoint PR`, here's 2 optional raw strategy from me:

1. hold a map to store the `ExecutionMode -> Concrete FlinkClientTrait`, then use mapping to judge the `Concrete FlinkClientTrait` to drive corresponding method. move the relation mapping between `xxxRequest` and `method name` into the concrete `xxxRequest` with  behavior.
In this way, if we introduce a new client action, we need   
-  add a `xxxRequest` first,
-  add a  xxxrequest->method name  mapping.
-  then, add a method declaration in `FlinkClientTrait` , 
- next,  add the corresponding method implementations in `xxxClient` with its' executionMode,  
- finally, add the same method for `FlinkClientHandler` ()
In other words, we only reduce the redundant case lines in every method of `FlinkClientHandler` by this way. but can't follow the OCP rule well.

2.  Introduce  a abstract base class that has `identifier` behavior to represent the method name and a `getFlinkClientTrait`  behavior to represent the mapping from `execution mode` to `Concrete FlinkClientTrait`.
And make `FlinkClientHandler` as a child class of `FlinkClientTrait`,which is helpful to unify the interface about Client classes. 
In this way, if we introduce a new client action, we need   
-  add a `xxxRequest` first, we need add the required implementations from the above abstract parent, 
-  then, add a method declaration in `FlinkClientTrait` , 
- next,  add the corresponding method implementations in all of children object difinetion.

We can reduce the redundant case lines in every method of `FlinkClientHandler` by this way and follow the OCP rule better. However, the `method name` attribute of xxxRequest used to drive in the xxxClient is a little weird for me.

If I'm wrong, correct me, please. Any suggestion is appreciated.
Thx ~

Reply:
> There's some determinate condition in the client.
> 
> 1. The ExecutionMode field is required for every existed XXXRequest and behavior like these lines.
> 2. every request need a method to drive the action bounded with `xxxClient`
> 3. special ExecutionMode with special xxxClient.
> 
> If we want to remove the case lines , which was mentioned in the `trigger savepoint PR`, here's 2 optional raw strategy from me:
> 
> 1. hold a map to store the `ExecutionMode -> Concrete FlinkClientTrait`, then use mapping to judge the `Concrete FlinkClientTrait` to drive corresponding method.
>    In this way, if we introduce a new client action, we need
> 
> * add a `xxxRequest` first,
> * add a  xxxrequest->method name  mapping.
> * then, add a method declaration in `FlinkClientTrait` ,
> * next,  add the corresponding method implementations in `xxxClient` with its' executionMode,
> * finally, add the same method for `FlinkClientHandler` ()
>   In other words, we only reduce the redundant case lines in every method of `FlinkClientHandler` by this way. but can't follow the OCP rule well.
> 
> 2. Introduce  a abstract base class that has `identifier` behavior to represent the method name and a `getFlinkClientTrait`  behavior to represent the mapping from `execution mode` to `Concrete FlinkClientTrait`.
>    And make `FlinkClientHandler` as a child class of `FlinkClientTrait`,which is helpful to unify the interface about Client classes.
>    In this way, if we introduce a new client action, we need
> 
> * add a `xxxRequest` first, we need add the required implementations from the above abstract parent,
> * then, add a method declaration in `FlinkClientTrait` ,
> * next,  add the corresponding method implementations in all of children object difinetion.
> 
> We can reduce the redundant case lines in every method of `FlinkClientHandler` by this way and follow the OCP rule better. However, the `method name` attribute of xxxRequest used to drive in the xxxClient is a little weird for me.
> 
> If I'm wrong, correct me, please. Any suggestion is appreciated. Thx ~

Thanks for your feedback and comments, updated .